staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go

```diff
@@ -309,6 +309,26 @@ func IsValidIP(value string) []string {
 	return nil
 }
 
+// IsValidIPv4Address tests that the argument is a valid IPv4 address.
+func IsValidIPv4Address(fldPath *field.Path, value string) field.ErrorList {
+	var allErrors field.ErrorList
+	ip := net.ParseIP(value)
+	if ip == nil || ip.To4() == nil {
+		allErrors = append(allErrors, field.Invalid(fldPath, value, "must be a valid IPv4 address"))
+	}
+	return allErrors
+}
+
+// IsValidIPv6Address tests that the argument is a valid IPv6 address.
+func IsValidIPv6Address(fldPath *field.Path, value string) field.ErrorList {
+	var allErrors field.ErrorList
+	ip := net.ParseIP(value)
+	if ip == nil || ip.To4() != nil {
```

Comment:
same as above.

Reply:
Will make a note of updating this in the future, but since the methods I'd actually need here only partially exist in netutils (no IPv4 methods), I rolled back to the initial implementation here. Will make a note of filing a PR against `k/utils` in the future, but it didn't seem worth holding up this PR for another one since there are other PRs already waiting for this to get in.